Lib/fontmake/font_project.py

```diff
@@ -435,17 +436,24 @@ def subset_otf_from_ufo(self, otf_path, ufo):
         keep_glyphs = set(ufo.lib.get(GLYPHS_PREFIX + 'Keep Glyphs', []))
 
         include = []
-        ufo_order = [glyph_name
-                     for glyph_name in ufo.lib[PUBLIC_PREFIX + 'glyphOrder']
-                     if glyph_name in ufo]
-        for old_name, new_name in zip(
-                ufo_order,
-                TTFont(otf_path).getGlyphOrder()):
+        ufo_order = makeOfficialGlyphOrder(ufo)
+        ot_order = TTFont(otf_path).getGlyphOrder()
+        assert ot_order[0] == ".notdef", (
+            "{}, subsetting: .notdef must be the first glyph in "
+            "{}.".format(ufo.path, otf_path)
```

Comment:
this assert is for the devs' own consumption, not the user's. I think the error message is too much in this case